### PR TITLE
Create a copy of upstream images we customize

### PR DESCRIPTION
**Pull request by kubasobon**
Towards https://github.com/giantswarm/roadmap/issues/3059

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -138,6 +144,14 @@ func (img *CustomImage) RetagUsingSHA() error {
 			return fmt.Errorf("finished %q with %d errors", img.Image, errorCount)
 		}
 		return nil
+	} else {
+		source := fmt.Sprintf("%s%s@sha256:%s", dockerTransport, img.Image, img.SHA)
+		wg := sync.WaitGroup{}
+		wg.Add(1)
+		// AzureCR/upstream
+		destination := fmt.Sprintf("%s%s/%s:@sha256:%s", dockerTransport, azureUpstreamURL, upstreamName, img.SHA)
```
Comment (piontec) [inline main.go:152]: Again, shouldn't this be tag, not SHA?

File: main.go [hunk 2]
```diff
@@ -265,20 +280,31 @@ tagLoop:
...
-			wg.Add(3)
+			wg.Add(4)
 			// Quay
 			destination := fmt.Sprintf("%s%s/%s:%s", dockerTransport, quayURL, destinationName, destinationTag)
 			go copyImage(&wg, errorCounter, source, destination)
 			// Azure
 			destination = fmt.Sprintf("%s%s/%s:%s", dockerTransport, azureURL, destinationName, destinationTag)
 			go copyImage(&wg, errorCounter, source, destination)
+			// Azure upstream
+			destination = fmt.Sprintf("%s%s/%s:%s", dockerTransport, azureUpstreamURL, upstreamName, tag)
```
Comment (piontec) [inline main.go:291]: `destinationTag` - the modified one?

File: main.go [hunk 3]
```diff
@@ -265,20 +280,31 @@ tagLoop:
...
 			wg.Wait()
 
 			// We'll skip to the next tag
 			continue
+		} else {
+			source := fmt.Sprintf("%s%s:%s", dockerTransport, img.Image, tag)
+			wg := sync.WaitGroup{}
+			wg.Add(1)
+			// AzureCR/upstream
+			destination := fmt.Sprintf("%s%s/%s:%s", dockerTransport, azureUpstreamURL, upstreamName, tag)
```
Comment (piontec) [inline main.go:305]: Same, shouldn't this use `destinationTag`? Or if we want to keep the original upstream tag without potential suffix changes, shouldn't we assign both to indicate that these are the same images?

**Comment by marians** [comment]
This was an attempt. Closing this, as we are currently not pursuing it.

**Pull request closed without merging**